weaknesses/MASVS-RESILIENCE/MASWE-0098.md

```diff
@@ -2,17 +2,43 @@
 title: App Virtualization Environment Detection Not Implemented
 id: MASWE-0098
 alias: app-virtualization-detection
-platform: [android, ios]
+platform: [android]
 profiles: [R]
 mappings:
-  masvs-v2: [MASVS-RESILIENCE-1]
-
-draft:
-  description: runs as a so-called "clone app"
-  topics:
-  - detection in place
-  - Effectiveness Assessment (e.g. bypassing the detection)
+  masvs-v1: [MSTG-RESILIENCE-3, MSTG-RESILIENCE-4, MSTG-RESILIENCE-5, MSTG-RESILIENCE-6]
+  masvs-v2: [MASVS-RESILIENCE-1, MASVS-RESILIENCE-4]
 status: draft
-
 ---
 
+## Overview
+
+Android app virtualization software is a potential vector for dynamic analysis and application tampering. Such software can partially or fully virtualize the Android system, enabling target applications to run in a controlled environment. It bypasses the Android security model by creating a more permissive environment that other threat tools can exploit, often without requiring the device to be rooted.
+
+Malware can exploit Android virtualization software by encapsulating legitimate applications within a virtualized environment, creating seemingly identical user experience. This kind of virtualization allows malicious actors to inject code that bypasses security measures, such as anti-tampering protections, enabling malware to execute fraudulent activities.
+
+Dynamic instrumentation tools, like [GameGuardian](https://gameguardian.net/forum/topic/20322-gameguardian-work-without-root/), can leverage Android virtualization software to operate on non-rooted devices.
+
+## Impact
+
+Maliciously used Android virtualization software can lead to a variety of security issues, including but not limited to:
+
+- **Unauthorized reading of application private data**: Malware utilizing Android virtualization software to virtualize a target application can access its private data. This can lead to exposure of sensitive user information, such as login credentials, financial data, and personal messages, compromising user privacy and security.
+- **Unauthorized alteration of application private data**: By injecting malicious code through virtualization, malware can modify private data within an application. This alteration can result in unauthorized transactions, changes to user settings, or tampering with the application's functionality, potentially causing financial loss or operational disruptions.
+- **Unauthorized escalation of privileges or identity theft**: Virtualization software can create a more permissive environment, enabling the execution of other malicious tools that would normally require root permissions.
+- **Circumventing security mechanisms**: Virtualizers can dynamically tamper with the application's memory without altering the application's integrity. This allows malicious code to bypass certain anti-tampering security measures.
+
+## Modes of Introduction
```

Comment:
This section seems to extend and overlap with the description of the weakness. It should focus on the ways in which an application can introduce this weakness. For example, the use of certain libraries or APIs, the declaration (or lack thereof) of certain attributes or permissions, etc. Anything that indicates that the app can be run inside a virtualization environment, or "on-device container" as Android calls it.

Specific example: not opting out of on-device Android container apps via the REQUIRE_SECURE_ENV manifest flag.

We'll use this information later to define the tests.